ts/features/idpay/onboarding/xstate/machine.ts

```diff
@@ -285,8 +347,57 @@ const createIDPayOnboardingMachine = () =>
                 target: "DISPLAYING_INITIATIVE"
               }
             ]
+          },
+          onDone: {
+            target: "ACCEPTING_REQUIRED_CRITERIA"
+          },
+          states: {
+            EVALUATING_SELF_CRITERIA: {
+              tags: [LOADING_TAG],
+              always: [
+                {
+                  target: "DISPLAYING_BOOL_CRITERIA",
+                  cond: "hasBoolRequiredCriteria"
+                },
+                {
+                  target: "CYCLE_MULTI_CRITERIA",
+                  cond: "hasMultiRequiredCriteria"
+                }
+              ]
+            },
+            CYCLE_MULTI_CRITERIA: {
+              tags: [WAITING_USER_INPUT_TAG],
+              entry: "navigateToMultiSelfDeclarationsScreen",
+              on: {
+                ADD_SELF_CONSENT: {
+                  actions: "addSelfConsent"
+                },
+                ALL_CRITERIA_ACCEPTED: {
+                  target: "ALL_CRITERIA_ACCEPTED"
+                }
+              }
+            },
+            DISPLAYING_BOOL_CRITERIA: {
+              tags: [WAITING_USER_INPUT_TAG],
+              entry: "navigateToBoolSelfDeclarationsScreen",
+              on: {
+                ACCEPT_REQUIRED_BOOL_CRITERIA: [
+                  {
+                    cond: "hasMultiRequiredCriteria",
+                    target: "CYCLE_MULTI_CRITERIA"
+                  },
+                  {
+                    target: "ALL_CRITERIA_ACCEPTED"
+                  }
+                ]
+              }
```

Comment:
I think that this type of decision should be made by the machine itself and not by the UI. 
We should trigger always the same event, then the machine should decide wether to go to the next criteria or to finish the flow.

Something like this:

```suggestion
               on: {
                ACCEPT_MULTI_CRITERIA: [
                  {
                    actions: "acceptMultiCriteria",
                    cond: "hasMoreCriteria"
                  },
                  {
                    actions: "acceptMultiCriteria",
                    target: "ALL_CRITERIA_ACCEPTED"
                  }
                ]
              }
```

Reply:
CONTEXT: the onboarding's stack navigator resets the machine on `navigation.push`, so we are forced to use the screen's navigation props.

Since we are forced to handle the navigation in the screen or rather in a custom hook (due to the huge amount of new logic that passing a second navigator to the machine would add), moving the check to the machine could possibly force the user to quit the cycle in case for some reason an entry of the answers array was left empty due to a possible bug, while handling the case in a custom hook allows us to let the user check their choices again.